modules/core/src/v2/coins/xlm.ts

```diff
@@ -597,11 +606,11 @@ export class Xlm extends BaseCoin {
         throw new Error('Provided wallet has a balance of 0 XLM, recovery aborted');
       }
 
-      const walletBalance = this.bigUnitsToBaseUnits(nativeBalanceInfo.balance);
-      const minimumReserve = yield this.getMinimumReserve();
-      const baseTxFee = yield this.getBaseTransactionFee();
+      const walletBalance = Number(self.bigUnitsToBaseUnits(nativeBalanceInfo.balance));
```

Comment:
do we have to worry about overflow here (i.e. use BigNumbers?)

Reply:
So funny story. `bigUnitsToBaseUnits` returns a string, `getMinimumReserve` returns a number, and so does `getBaseTransactionFee`. So prior to this change the types in the expression which assigns `recoveryAmount` are `string - number - number`. And that works, because javascript.

This change just makes it `number - number - number`, which has all the same overflow problems as the previous implementation, but at least fixes the mixed types.